src/graph/graphcontext.c

```diff
@@ -54,10 +54,11 @@ GraphContext* GraphContext_New(RedisModuleCtx *ctx, const char *graphname,
   return gc;
 }
 
-GraphContext* GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphname) {
+GraphContext* GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphname, bool readOnly) {
   GraphContext *gc = NULL;
   RedisModuleString *rs_name = RedisModule_CreateString(ctx, graphname, strlen(graphname));
-  RedisModuleKey *key = RedisModule_OpenKey(ctx, rs_name, REDISMODULE_READ);
+  int readWriteFlag = readOnly ? REDISMODULE_READ : REDISMODULE_WRITE;
```

Comment:
The flags that build the mode argument are bitmasked, so I think in the write case we probably want to specify `REDISMODULE_READ|REDISMODULE_WRITE`.